pkg/auth/auth_test.go

```diff
@@ -33,6 +33,8 @@ import (
 
 type fakeK8sAuth struct {
 	DiscoveryCli discovery.DiscoveryInterface
+	canIResult   bool
+	canIError    error
```

Comment:
This test change was while I was first created unit tests for the new ValidateForNamespace before realising it's a one-line function proxying to the existing `CanI` function, but I left the changes as it's more extensible, allowing individual tests to decide what the result of the `CanI` call should be.